creusot/src/gather_spec_closures.rs

```diff
@@ -12,11 +12,10 @@ use rustc_middle::{
     mir::{visit::Visitor, AggregateKind, BasicBlock, Body, Location, Operand, Rvalue},
     ty::{TyCtxt, TyKind},
 };
-use rustc_span::Symbol;
 
 #[derive(Debug, Clone, Copy)]
 pub enum LoopSpecKind {
-    Invariant(Symbol),
```

Comment:
So I had left named invariants to give people time to upgrade their why3, I figured we would start by also deprecating it in Creusot.

Reply:
Isn't it already backward compatible wrt. why3 ? I thought Why3 supported unnamed invariants in the past.

Since when do we deprecate things in Creusot? That would really be the first time!

Anyways, it's really not a difficult change to do on the code. Simply remove invariant names.

Reply:
Mlcfg invariants needed to be named until yesterday, and only with the master branch of why3. The reason I've noticed is that we have external users will do an opam install why3 and it won't work. Another thing I was thinking is that cargo Creusot could try to detect the version of why3 and issue a warning if it's too old.